### ISSUE TITLE
Connection error: websocket: bad handshake

### ISSUE THREAD

**Issue by GVRGowtham** [jpillora/chisel#82]
I am using the latest chisel binaries released on January, trying to connect to Predix database without any proxy. It throws up connection error and keeps retrying though it fails every single time. The system I tried is Mac with High Sierra OS.

However, the client runs fine on Windows machine and I am able to connect to the same database without any issues.

Error logs:

2019/02/27 21:46:46 client: Connecting to wss://predix-chisel-postgres-theosophic-madam.run.aws-usw02-pr.ice.predix.io:443
2019/02/27 21:46:46 client: proxy#1:0.0.0.0:4006=>db-ac-***:5432: Listening
2019/02/27 21:46:47 client: Connection error: websocket: bad handshake
2019/02/27 21:46:47 client: Retrying in 100ms...
2019/02/27 21:46:50 client: Connection error: websocket: bad handshake (Attempt: 1)
2019/02/27 21:46:50 client: Retrying in 200ms...
2019/02/27 21:46:51 client: Connection error: websocket: bad handshake (Attempt: 2)
2019/02/27 21:46:51 client: Retrying in 400ms...
2019/02/27 21:46:53 client: Connection error: websocket: bad handshake (Attempt: 3)
2019/02/27 21:46:53 client: Retrying in 800ms...
2019/02/27 21:46:55 client: Connection error: websocket: bad handshake (Attempt: 4)
2019/02/27 21:46:55 client: Retrying in 1.6s...
2019/02/27 21:46:59 client: Connection error: websocket: bad handshake (Attempt: 5)
2019/02/27 21:46:59 client: Retrying in 3.2s...
2019/02/27 21:47:04 client: Connection error: websocket: bad handshake (Attempt: 6)
2019/02/27 21:47:04 client: Retrying in 6.4s...
2019/02/27 21:47:11 client: Connection error: websocket: bad handshake (Attempt: 7)
2019/02/27 21:47:11 client: Retrying in 12.8s...
2019/02/27 21:47:25 client: Connection error: websocket: bad handshake (Attempt: 8)
2019/02/27 21:47:25 client: Retrying in 25.6s...
2019/02/27 21:47:52 client: Connection error: websocket: bad handshake (Attempt: 9)
2019/02/27 21:47:52 client: Retrying in 51.2s...

**Comment by jpillora** [comment]
Are both sides running the same version?

**Comment by GVRGowtham** [comment]
I'm not sure of that. Its the chisel commonly used across the developers to connect to Predix database

**Comment by GVRGowtham** [comment]
Thanks @jpillora I redeployed the chisel using the latest binaries to my server. Thanks, it works now 👍

**Comment by fanyang-ceg** [comment]
Tried on win10 (version 1909); go version: 1.14.4; Chisel 0.0.0-src, but still experiencing the "client: Connection error: websocket: bad handshake (Attempt:#)" error

**Comment by jpillora** [comment]
Can you put the latest RC version on both ends? Also are verbose logs on?

**Comment by fanyang-ceg** [comment]
I first used the compiled version (%userprofile%\go\bin\chisel.exe, which is generated by " go get -v github.com/jpillora/chisel"), that one failed with bad handshake.



The server was using the (Linux) binary from https://github.com/briangann/predix-chisel-postgres (deployed to cloud foundry); and my local is windows, probably the versions don't match.



I retried the windows executable from predix-chisel-postgres repo, it finally worked.



Great thanks for your quick response.



Fan
